app/main.py

```diff
@@ -1 +1,52 @@
-# write your code here
+class SoftwareEngineer:
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        if "skills" not in self.__dict__:
+            self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+
+    def __init__(self, name: str) -> None:
+        if "skills" in self.__dict__:
+            self.skills += ["JavaScript", "HTML", "CSS"]
+        else:
+            self.skills = ["JavaScript", "HTML", "CSS"]
```

Comment:
The `FrontendDeveloper` class attempts to initialize `skills` before calling `super().__init__(name)`. This can lead to issues because the `SoftwareEngineer` constructor is not called first, which is generally expected in Python's class inheritance. Consider calling `super().__init__(name)` before setting the `skills` attribute.